### PR TITLE
Attestations: Email/Phone/Facebook/Twitter

### PR DESCRIPTION
**Pull request by tyleryasaka**
### Checklist:

- [x] Code contains relevant tests for the problem you are solving
- [x] Ensure all new and existing tests pass
- [x] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/docs)
- [x] Submit to the `develop` branch instead of `master`

### Description:

After this PR, I will work on re-defining our verification API using a new library; see https://github.com/OriginProtocol/bridge-server/issues/21

1. This updates the phone verification to return an actual signature for the attestation. The signature is compatible with the current implementation of ERC725 that we're using.
2. Adds email verification using Sendgrid
3. Adds Facebook & Twitter verification using Oauth

Finishes https://github.com/OriginProtocol/bridge-server/issues/22

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -63,4 +67,5 @@ urllib3==1.22
 waitress==1.1.0
 web3==4.1.0
 wcwidth==0.1.7
+web3==4.1.0
```
Comment (Gzing) [inline requirements.txt]: Duplicate entry, please remove

File: testing/test_data.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from dateutil import parser as dateparser
```
Comment (Gzing) [inline testing/test_data.py]: Do we need this file anymore with the `factories` in place?

File: logic/verification_service.py [hunk 3]
```diff
@@ -73,8 +95,148 @@ def verify_phone(self, req):
...
+            web3, signing_key, req.eth_address, claim_type, data)
         return verification.VerifyPhoneResponse(
-            attestation=generate_signed_attestation(addr))
+            signature=signature, claim_type=claim_type, data=data)
+
+    def generate_email_verification_code(self, req):
+        addr = numeric_eth(req.eth_address)
+        db_code = VC.query \
+            .filter(VC.eth_address == addr) \
+            .filter(VC.email == req.email) \
```
Comment (Gzing) [inline logic/verification_service.py]: @tyleryasaka Should we query email case insensitively here and other places with similar query definition?

Comment (tyleryasaka) [reply logic/verification_service.py]: Yeah, good call. Technically email addresses can be case sensitive before the `@`, but in practice they are always case insensitive. So it's a safe assumption. https://stackoverflow.com/a/9808332

Done 👍

File: testing/test_data.py (hunk 2 above)
Comment (tyleryasaka) [reply testing/test_data.py]: No we don't. Removed. 👍

File: requirements.txt (hunk 1 above)
Comment (tyleryasaka) [reply requirements.txt]: Done 👍

**Comment by tyleryasaka** [comment]
All of these commits are in https://github.com/OriginProtocol/bridge-server/pull/28 so I'm going to close this and just use that branch. It's a lot of code but it's been reviewed pretty well at this point.

**Pull request closed without merging**